File: go/binary_search/count.go
package binary_search

// CountInts should return number of values in a sorted array
func CountInts(arr []int, val int) int {
	l, r := 0, len(arr)
	for l < r {
		m := (l + r) / 2
		if arr[m] < val {
			l = m + 1
		} else {
			r = m
		}
	}

	l0 := l
	r = len(arr)
	for l < r {
		m := (l + r) / 2
		if arr[m] == val {
			l = m + 1
		} else {
			r = m
		}
	}

	return r - l0
}

File: python/binary_search/count.py

# count_ints should return number of value occurrences in a sorted array
def count_ints(arr, val: int) -> int:
	# raise NotImplementedError('TODO')

	l, r = 0, len(arr)
	while l < r:
		m = (l + r) // 2
		if arr[m] < val:
			l = m+1
		else:
			r = m

	l0 = l
	l, r = 0, len(arr)
	while l < r:
		m = (l + r) // 2
		if arr[m] <= val:
			l = m+1
		else:
			r = m

	return r - l0
